week3/average/kirikaza/Average.hs

```diff
@@ -0,0 +1,24 @@
+module Main where
+
+import Control.Monad (forM_)
+import Data.Function (on)
+import qualified Data.Sequence as Seq
```

Comment:
I'm not sure it'll actually be faster (thanks to `Seq`'s constant factors). Also, see @vitcra's solution for several alternative implementations, some of which are pretty nice